src/compiler/crystal/compiler.cr

```diff
@@ -258,13 +258,18 @@ module Crystal
       @optimization_mode.o3? && @single_module
     end
 
+    def single_module_compilation?
```

Comment:
Note: I didn't call that method `single_module?` because there already is a `getter? single_module : Bool` that is used to create a host compiler in macros (I believe it's for the `run` macro).

Reply:
Hm, this seems not ideal 🤔

Reply:
Wait, what are your talking about. There's `property? single_module : Bool` which reflects the `--single-module` flag.

https://github.com/crystal-lang/crystal/blob/8823b75a4570101c94584f4f70c5feea48836db0/src/compiler/crystal/command.cr#L547-L549

I suppose that's more or less what we want + cross compile and emit targets. But maybe those should be accounted for in that ivar as well?

Reply:
Oh and then we forward that flag to https://github.com/crystal-lang/crystal/blob/8823b75a4570101c94584f4f70c5feea48836db0/src/compiler/crystal/macros/macros.cr#L137
Dunno why that is though? The host compiler for macros should not necessarily inherit the entire target compiler configuration...
I can't really make sense of this.

Reply:
Then let's at least rename the current `single_module` to `single_module_flag?` and put a pin to return to this later.

Reply:
I refactored. We don't need to copy the single module flag, and it's overriden by the call to `#release!` anyway. I also added a couple methods to set `@single_module` along with cross compile and adding emit targets so we don't end up with a weird `@single_module == false` but `#single_module?` would be true.